crates/sierra-casm/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+use cairo_lang_utils::bigint::BigUintAsHex;
+use num_bigint::BigUint;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use serde_json::Error;
+use starknet::core::types::FlattenedSierraClass;
+
+use cairo_lang_starknet::casm_contract_class::CasmContractClass;
+use cairo_lang_starknet::contract_class::{ContractClass, ContractEntryPoints};
+use cairo_lang_starknet_v1_0_0_alpha6::casm_contract_class::CasmContractClass as CasmContractClassV1Alpha;
+use cairo_lang_starknet_v1_0_0_alpha6::contract_class::ContractClass as ContractClassV1Alpha;
+use cairo_lang_starknet_v1_0_0_rc0::casm_contract_class::CasmContractClass as CasmContractClassV1Rc0;
+use cairo_lang_starknet_v1_0_0_rc0::contract_class::ContractClass as ContractClassV1Rc0;
+use cairo_lang_starknet_v1_1_1::casm_contract_class::CasmContractClass as CasmContractClassV1_1;
+use cairo_lang_starknet_v1_1_1::contract_class::ContractClass as ContractClassV1_1;
+
+pub fn compile(definition: &FlattenedSierraClass) -> anyhow::Result<CasmContractClass> {
+    macro_rules! compile_contract {
+        ($sierra_type:ty, $casm_type:ty) => {{
+            let sierra_class = flattened_sierra_to_contract_class::<$sierra_type>(definition)?;
+            let maybe_casm_class = <$casm_type>::from_contract_class(sierra_class, true);
+            if let Ok(casm_class) = maybe_casm_class {
+                return casm_to_main_casm::<$casm_type>(&casm_class);
+            }
+        }};
+    }
+
+    compile_contract!(ContractClass, CasmContractClass);
+    compile_contract!(ContractClassV1_1, CasmContractClassV1_1);
+    compile_contract!(ContractClassV1Rc0, CasmContractClassV1Rc0);
+    compile_contract!(ContractClassV1Alpha, CasmContractClassV1Alpha);
+
+    unreachable!()
```

Comment:
I think we settled on not using the unreachable macro where it can be a panic (i think this case can even be a handleable error since we are extracting it out to a crate)